File: server/Model/DTOs/Base/CursorPaginatedResultDto.cs
// server/Model/DTOs/Base/CursorPaginatedResultDto.cs
namespace Model.DTOs.Base;

public class CursorPaginatedResultDto<T,TCursor>
{
    public List<T> Items { get; set; } = new();
    public TCursor? NextCursor { get; set; }
    public bool HasMore { get; set; }
}

File: client/types/base/CursorPaginatedResultDto.ts
export type CursorPaginatedResultDto<T, TCursor> = {
  nextCursor: TCursor;
  hasMore: boolean;
  items: T[];
};

export const defaultCursorPaginatedResult: CursorPaginatedResultDto<any, any> = {
  nextCursor: null,
  hasMore: false,
  items: [],
};
